src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowRunTest.java

```diff
@@ -491,4 +498,208 @@ private void assertCulprits(WorkflowRun b, String... expectedIds) throws IOExcep
         assertEquals(1, checkouts.size());
         assertEquals(GitSCM.class, checkouts.get(0).getClass());
     }
+
+    @Test
+    @Issue("NGPIPELINE-917")
+    // Test proves that turning on changelog/polling and then turning it off again will remove the polling baseline
+    public void baselineResetSingleRepo() throws Exception {
+        sampleRepo.init();
+        TaskListener listener = StreamTaskListener.fromStdout();
+        //Creating the job
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        //Setting definition with changelog and polling true
+        p.setDefinition(new CpsFlowDefinition(
+                "node {\n" +
+                checkoutString(sampleRepo, true, true) +
+                "}\n", false));
+
+        WorkflowRun b1 = p.scheduleBuild2(0).waitForStart();
+        r.assertBuildStatusSuccess(r.waitForCompletion(b1));
+        assertThat(b1.checkouts(listener).size(), equalTo(1));
+        // Checking that the polling baseline is correctly established
+        assertThat(b1.checkouts(listener).get(0).pollingBaseline, notNullValue());
+
+        //Resetting the definition to turn off changelog and polling
+        p.setDefinition(new CpsFlowDefinition(
+                "node {\n" +
+                checkoutString(sampleRepo, false, false) +
+                "}\n", false));
+
+
+        WorkflowRun b2 = p.scheduleBuild2(0).waitForStart();
+
+        r.assertBuildStatusSuccess(r.waitForCompletion(b2));
+        assertThat(b2.checkouts(listener).size(), equalTo(1));
+        //proving the baseline is removed
+        assertThat(b2.checkouts(listener).get(0).pollingBaseline, nullValue());
+    }
+
+    @Test
+    @Issue("NGPIPELINE-917")
+    // Showing how only with both changelog/poll false is when it removes the checkout
+    public void baselineResetSingleRepoEdgeCases() throws Exception {
```

Comment:
@dwnusbaum So this is eventually what I would like to start failing because this is not the ideal scenario. Ideally you could actually turn off just changelog and it would turn off both so I kinda wanted a base case and then more nuanced to reveal the difference in behavior.

I am OK with showing both here I just wanted to see if that changed your perspective

Reply:
> Ideally you could actually turn off just changelog and it would turn off both so I kinda wanted a base case and then more nuanced to reveal the difference in behavior. 

I don't think we could make that change directly, since it might be a breaking change for some users, so I think the current behavior is here to stay.

Some things I think we could do in workflow-scm-step to help users:

* Log a warning if a users specifies `changelog: true` and `poll: false` explaining that `poll: false` is overriden by `changelog: true`
* Offer an alternate parameter, like `mode` or something, and have 3 options: `CHANGELOG_AND_POLL`, `POLL_ONLY`, `NONE` (bad names but you get the idea), so that users don't have a way to set `changelog: true` and `poll: false`. Probably not worth the complexity since we'd still need to support `changelog` and `poll` in addition to the new option.